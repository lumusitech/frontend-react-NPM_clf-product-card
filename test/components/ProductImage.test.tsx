import React from 'react';

import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('should shows component with the custom img', () => {
    const wrapper = renderer.create(<ProductImage img="imagen.png" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should shows component with the product image', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
