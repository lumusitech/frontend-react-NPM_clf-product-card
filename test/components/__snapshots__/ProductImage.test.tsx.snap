// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should shows component with the custom img 1`] = `
<img
  alt="Product img"
  className="productImg undefined"
  src="imagen.png"
/>
`;

exports[`ProductImage should shows component with the product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product img"
    className="productImg undefined"
    src="./coffe-mug.png"
  />
</div>
`;
